src/Domain/Model/Restaurant.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Model;
+
+use App\Domain\Migrations\TablesNameCatalog;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Uid\Uuid;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="TablesNameCatalog::RESTAURANTS")
+ */
+class Restaurant
+{
+    /**
+     * @ORM\Id()
+     * @ORM\Column(type="string", unique=true)
+     * @Assert\Uuid
+     *
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string", length=150)
+     * @Assert\NotBlank()
+     *
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="string", length=150)
+     * @Assert\NotBlank()
+     *
+     * @var string
+     */
+    private $alias;
+
+    /**
+     * @ORM\Column(type="string", length=255, nullable=true)
+     * @Assert\Url
+     *
+     * @var string
+     */
+    private $url;
+
+    /**
+     * @ORM\Column(type="array", nullable=true)
+     *
+     * @var array
+     */
+    private $categories = [];
+
+    /**
+     * @ORM\Column(type="string", length=150, nullable=true)
+     *
+     * @var string
+     */
+    private $location;
+
+    /**
+     * @ORM\Column(type="string", length=150, nullable=true)
+     *
+     * @var string
+     */
+    private $coordinates;
+
+    /**
+     * @ORM\Column(type="string", length=20, nullable=true)
+     * @PhoneNumber()
+     *
+     * @var string
+     */
+    private $phoneNumber;
+
+    /**
+     * @ORM\Column(type="datetime_immutable")
+     * @Assert\NotNull()
+     *
+     * @var DateTime
+     */
+    private $created_at;
+
+    /**
+     * @ORM\Column(type="datetime")
+     * @Assert\NotNull()
+     *
+     * @var DateTime
+     */
+    private $updated_at;
+
+    /**
+     * @ORM\Column(type="datetime", nullable=true)
+     *
+     * @var DateTime
+     */
+    private $deleted_at;
+
+    public function __construct()
+    {
+        $this->id = Uuid::v4();
+        $this->createdAt = new \DateTimeImmutable();
+        $this->updatedAt = new \DateTimeImmutable();
```

Comment:
Suggestion : Tu peux donner un new DateTime sur l'event PreUpdate doctrine :)

Reply:
I will use it on my setter like following, but if you have better solution i'll take it ^^
```
/**
 * @ORM\PrePersist
 * @ORM\PreUpdate
 */
public function setUpdatedAt()
```